### PR TITLE
docs: adding new tutorial on how to handle heaptrack with nim waku

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
As per the issue "[Heap analysis using heaptrack](https://github.com/waku-org/nwaku/issues/986)" we are adding a new tutorial that explains how to interact with heaptrack and nimwaku

Heaptrack is a memory analysis tool very useful to get a good insight of the underlying memory usage.

In this tutorial we describe the next "how to's":
- get and build locally the heaptrack tool (Ubuntu.).
- generate a heaptrack memory report.
- analyse the resulting report.
- create docker image with both heaptrack plus wakunode2

### CODE REVIEW

File: docs/tutorial/heaptrack.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+   - `sudo apt install libkf5widgetsaddons-dev`
+   - `sudo apt install libqt5gui5`
+   - `sudo apt install libkf5kio-dev`
+   - `sudo apt install libkf5iconthemes-dev`
+- `make`
+- On completion, the `bin/heaptrack_gui` and `bin/heaptrack` binaries will get generated.
+    - heaptrack: needed to generate the report.
+    - heaptrack_gui: needed to analyse the report.
+
+## Heaptrack & Nim Waku
```
Comment (jm-clius) [inline docs/tutorial/heaptrack.md:40]:
```suggestion
## Heaptrack & Nwaku
```

We can refer to it simply as "nwaku". :)

File: docs/tutorial/heaptrack.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+The make command should have the 'NIM_COMMIT' setting as:
+
+`make -j<nproc> NIM_COMMIT=heaptrack_support ...`
+
+This is to force the `nimbus-build-system` to use the Nim compiler that points at the [heaptrack_support](https://github.com/status-im/nim/tree/heaptrack_support) branch.
+
+### Create nwaku memory report with heaptrack
+
+nwaku only works correctly with heaptrack operating in 'preload' mode, i.e. the memory is tracked from the beginning. To achive this, the `heaptrack` binary should be prepended to the usual `wakunode`:
```
Comment (jm-clius) [inline docs/tutorial/heaptrack.md:53]:
```suggestion
nwaku only works correctly with heaptrack operating in 'preload' mode, i.e. the memory is tracked from the beginning.
To achieve this, the `heaptrack` binary should be prepended to the usual `wakunode`:
```

With [semantic breaks](https://sembr.org/) each sentence or standalone clause can be in a new line (single newlines are not rendered in markdown, but makes it easier to review docs).

**Review by jm-clius — APPROVED** [review]
Great work on this (both getting `heaptrack` working properly and documenting it in a tutorial)!🙌 Consider using semantic breaks between lines and clauses: https://sembr.org/

**Review by alrevuelta — APPROVED** [review]
great thanks for all the work with this!

**Pull request merged**